app/assets/javascripts/discourse/app/routes/topic.js

```diff
@@ -226,9 +226,11 @@ const TopicRoute = DiscourseRoute.extend({
 
   @action
   changeOwner() {
-    showModal("change-owner", {
-      model: this.modelFor("topic"),
-      title: "topic.change_owner.title",
+    this.modal.show(ChangeOwnerModal, {
+      model: {
+        topicController: this.controllerFor("topic"),
+        topic: this.modelFor("topic"),
+      },
```

Comment:
I realise its expanding the scope of this PR a little... but how feasible would it be to move this action out of the route? As far as I can tell, it's only called from the `selected-posts` component, so perhaps it could be moved inside that component's JS?

https://github.com/discourse/discourse/blob/a673004777005d7157710615736d3415f23c04e7/app/assets/javascripts/discourse/app/components/selected-posts.hbs#L44-L44

Reply:
Aha I see, let's keep this PR as-is then 👌